pkg/staticnetworkconfig/generator.go

```diff
@@ -90,40 +92,16 @@ func (s *StaticNetworkConfigGenerator) generateHostStaticNetworkConfigData(ctx c
 	return filesList, nil
 }
 
-func (s *StaticNetworkConfigGenerator) executeNMStatectl(ctx context.Context, hostYAML string) (string, error) {
-	err := s.sem.Acquire(ctx, 1)
+func (s *StaticNetworkConfigGenerator) validateNMStateConfig(hostYAML string) (string, error) {
```

Comment:
This is not just for validating the config it also actually generates the resulting config files.

I think we should name it better or maybe, because it's so small, we just move the contents to where it's used.
It's only two calls and it is only used in two places.

Reply:
I think there's enough here for it to remain as a function. I changed the function name to  better reflect what it is doing.